libs/infrastructure.py

```diff
@@ -36,67 +42,91 @@ def create_vars_file(self):
 
     def generate_configmap(self):
         client = boto3.client('iam')
         role_arn = client.get_role(RoleName='eks-node-role')['Role']['Arn']
-        with open("/tmp/{}/nodes_cm.yaml".format(self.timestamp), "w") as nodes_cm:
+        with open("{}/nodes_cm.yaml".format(self.tmp_root_path), "w") as nodes_cm:
             j2_env = Environment(loader=FileSystemLoader("/opt/templates/"),
                                  trim_blocks=True)
             gen_template = j2_env.get_template('nodes_cm.j2').render(aws_iam_role_eksnode_arn=role_arn)
             nodes_cm.write(gen_template)
 
-    def apply_node_auth_configmap(self):
+    def __apply_node_auth_configmap(self):
         self.generate_configmap()
         process = Popen(['kubectl', 'apply', "-f",
-                         "/tmp/{}/nodes_cm.yaml".format(self.timestamp)],
+                         "{}/nodes_cm.yaml".format(self.tmp_root_path)],
                         env=dict(os.environ,
                                  **{"KUBECONFIG": self.kube_config_file}),
                         stdout=PIPE, stderr=PIPE)
         stdout, stderr = process.communicate()
-        process.wait()
+        return process.wait()
 
     def install_kube(self):
-        os.mkdir("/tmp/{}".format(self.timestamp))
-        process = Popen(['terraform', 'workspace', 'new', self.cluster_name, self.working_dir], cwd=self.cwd,
+        os.mkdir(self.tmp_root_path)
+        _cmd_wksps = ['terraform', 'workspace', 'new', self.cluster_name, self.tf_working_dir]
+        yield "START: Creating Workspace: {}".format(_cmd_wksps), None
+        process = Popen(_cmd_wksps, cwd=self.cwd,
                         stdout=PIPE, stderr=PIPE)
         stdout, stderr = process.communicate()
         self.logger.info("Create namespace")
         time.sleep(2)
         self.logger.info("stdout id: {}".format(stdout))
         self.logger.info("stderr id: {}".format(stderr))
         process.wait()
-        self.create_vars_file()
-        process = Popen(['terraform',
-                         'apply',
-                         '-var-file=/tmp/{}/tfvars.tf'.format(self.timestamp),
-                         '-auto-approve',
-                         self.working_dir], cwd=self.cwd,
+        yield "RUNNING: Terraform workspace created: {}", None
+
+        yield "RUNNING: Creating Terraform vars", None
+        self.__create_vars_file()
+        yield "RUNNING: Terraform vars created", None
+        _cmd_apply = ['terraform',
+                      'apply',
+                      '-var-file={}/{}'.format(self.tmp_root_path, TF_VARS_FILE),
+                      '-auto-approve',
+                      self.tf_working_dir]
+        yield "RUNNING: Actually creating cloud. This may take time... {}".format(_cmd_apply), None
+        process = Popen(_cmd_apply, cwd=self.cwd,
                         stdout=PIPE, stderr=PIPE)
         stdout, stderr = process.communicate()
         time.sleep(10)
```

Comment:
time sleep is not a good idea as the process might take longer...

Reply:
sure, not this pr